File: src/TR/sh_deprecated.ts
const Queue = {} as table
declare namespace Trailers {
	function Init(this: void, ventity: VEntity): void
}
const EntityMeta = FindMetaTable("Entity") as any
EntityMeta.SimfIsTrailer = function (this: Entity) {
	return this.GetNWBool("simf_istrailer", false)
}
EntityMeta.GetCenterposition = function (this: Entity) {
	return this.GetNWVector("simf_centerpos", Vector(0, -100, 0))
}
EntityMeta.GetTrailerCenterposition = function (this: Entity) {
	return this.GetNWVector("simf_trailercenterpos", Vector(0, 0, 0))
}
EntityMeta.TrailerCanConnect = function (this: Entity) {
	return this.GetNWVector("simf_trailercancon", false)
}
EntityMeta.CarCanConnect = function (this: Entity) {
	return this.GetNWVector("simf_carcancon", false)
}
EntityMeta.GetLeftSignalEnabled = function (this: Entity) {
	return this.GetNWVector("simfs_leftsig", false)
}
EntityMeta.GetRightSignalEnabled = function (this: Entity) {
	return this.GetNWVector("simfs_rightsig", false)
}
EntityMeta.VehicleGetCanConnect = function (this: Entity) {
	return this.GetNWBool("simfs_veh_canconnect", false)
}
function StartQueueIfNotExists(this: void, ent: Entity) {
	Queue[ent as any] = Queue[ent as any] || {}
	timer.Simple(1, () => { // now i know why arrow functions exists :D
		if (IsValid(ent)) {
			if (Queue[ent as any] != null) {
				MsgC(Color(255, 255, 0), "TR: this vehicle uses old api\n")
				Trailers.Init({ ent: ent, inputPos: Queue[ent as any].trailerCenterposition, outputPos: Queue[ent as any].centerposition })
				Queue[ent as any] = null
			}
		}
	})
}
EntityMeta.SetSimfIsTrailer = function (this: Entity, bool: boolean) {
	ErrorNoHalt("TR: this vehicle uses old api")
	StartQueueIfNotExists(this)
	this.SetNWBool("simf_istrailer", bool)
}
EntityMeta.SetCenterposition = function (this: Entity, vector: Vector) {
	StartQueueIfNotExists(this)
	Queue[this as any].centerposition = vector
	this.SetNWVector("simf_centerpos", vector)
}
EntityMeta.SetTrailerCenterposition = function (this: Entity, vector: Vector) {
	StartQueueIfNotExists(this)
	Queue[this as any].trailerCenterposition = vector
	this.SetNWVector("simf_trailercenterpos", vector)
}
EntityMeta.SetTrailerCanConnect = function (this: Entity, bool: boolean) {
	this.SetNWBool("simf_trailercancon", bool)
}
EntityMeta.SetCarCanConnect = function (this: Entity, bool: boolean) {
	this.SetNWBool("simf_carcancon", bool)
}
EntityMeta.SetLeftSignalEnabled = function (this: Entity, bool: boolean) {
	this.SetNWBool("simfs_leftsig", bool)
}
EntityMeta.SetRightSignalEnabled = function (this: Entity, bool: boolean) {
	this.SetNWBool("simfs_rightsig", bool)
}
EntityMeta.VehicleSetCanConnect = function (this: Entity, bool: boolean) {
	this.SetNWBool("simfs_veh_canconnect", bool)
}
export = null

File: lua/TR/sh_deprecated.lua
local Queue = {}
local EntityMeta = FindMetaTable("Entity")
EntityMeta.SimfIsTrailer = function(self)
    return self:GetNWBool("simf_istrailer", false)
end
EntityMeta.GetCenterposition = function(self)
    return self:GetNWVector(
        "simf_centerpos",
        Vector(0, -100, 0)
    )
end
EntityMeta.GetTrailerCenterposition = function(self)
    return self:GetNWVector(
        "simf_trailercenterpos",
        Vector(0, 0, 0)
    )
end
EntityMeta.TrailerCanConnect = function(self)
    return self:GetNWVector("simf_trailercancon", false)
end
EntityMeta.CarCanConnect = function(self)
    return self:GetNWVector("simf_carcancon", false)
end
EntityMeta.GetLeftSignalEnabled = function(self)
    return self:GetNWVector("simfs_leftsig", false)
end
EntityMeta.GetRightSignalEnabled = function(self)
    return self:GetNWVector("simfs_rightsig", false)
end
EntityMeta.VehicleGetCanConnect = function(self)
    return self:GetNWBool("simfs_veh_canconnect", false)
end
local function StartQueueIfNotExists(ent)
    Queue[ent] = Queue[ent] or ({})
    timer.Simple(
        1,
        function()
            if IsValid(ent) then
                if Queue[ent] ~= nil then
                    MsgC(
                        Color(255, 255, 0),
                        "TR: this vehicle uses old api\n"
                    )
                    Trailers.Init({ent = ent, inputPos = Queue[ent].trailerCenterposition, outputPos = Queue[ent].centerposition})
                    Queue[ent] = nil
                end
            end
        end
    )
end
EntityMeta.SetSimfIsTrailer = function(self, bool)
    ErrorNoHalt("TR: this vehicle uses old api")
    StartQueueIfNotExists(self)
    self:SetNWBool("simf_istrailer", bool)
end
EntityMeta.SetCenterposition = function(self, vector)
    StartQueueIfNotExists(self)
    Queue[self].centerposition = vector
    self:SetNWVector("simf_centerpos", vector)
end
EntityMeta.SetTrailerCenterposition = function(self, vector)
    StartQueueIfNotExists(self)
    Queue[self].trailerCenterposition = vector
    self:SetNWVector("simf_trailercenterpos", vector)
end
EntityMeta.SetTrailerCanConnect = function(self, bool)
    self:SetNWBool("simf_trailercancon", bool)
end
EntityMeta.SetCarCanConnect = function(self, bool)
    self:SetNWBool("simf_carcancon", bool)
end
EntityMeta.SetLeftSignalEnabled = function(self, bool)
    self:SetNWBool("simfs_leftsig", bool)
end
EntityMeta.SetRightSignalEnabled = function(self, bool)
    self:SetNWBool("simfs_rightsig", bool)
end
EntityMeta.VehicleSetCanConnect = function(self, bool)
    self:SetNWBool("simfs_veh_canconnect", bool)
end
local ____exports = nil
return ____exports
